src/Kernel.EFSupport/Provider/IRepository.cs

```diff
@@ -0,0 +1,18 @@
+using LT.DigitalOffice.Kernel.Attributes;
+using Microsoft.AspNetCore.JsonPatch;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.Kernel.EFSupport.Provider;
+
+[AutoInject]
+public interface IRepository<T> where T : class
+{
+  Task<IQueryable<T>> GetAllAsync();
+  Task<T> GetAsync(Guid id);
+  Task<Guid> CreateAsync(T entity);
+  Task<T> EditAsync(JsonPatchDocument<T> patch);
```

Comment:
Why not Task<bool> EditAsync()? Is it better to return entity?

Reply:
it might be useful for frontend to return update entity after editing